mmv1/products/dataplex/Task.yaml

```diff
@@ -0,0 +1,487 @@
+    path: 'error'
+    message: 'message'
+autogen_async: true
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/dataplex/docs'
+  api: 'https://cloud.google.com/dataplex/docs/reference/rest'
+import_format: ['projects/{{project}}/locations/{{location}}/lakes/{{lake}}/tasks/{{task_id}}']
+iam_policy: !ruby/object:Api::Resource::IamPolicy
+  exclude: false
+  method_name_separator: ':'
+  parent_resource_attribute: 'task'
+  fetch_iam_policy_verb: :POST
+  import_format:
+    [
+      'projects/{{project}}/locations/{{location}}/lakes/{{lake}}/tasks/{{task_id}}',
+      '{{task_id}}',
+    ]
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    url_param_only: true
+    immutable: true
+    description: |
+      The location in which the task will be created in.
+  - !ruby/object:Api::Type::String
+    name: 'lake'
+    url_param_only: true
+    immutable: true
+    description: |
+      The lake in which the task will be created in.
+  - !ruby/object:Api::Type::String
+    name: 'taskId'
+    url_param_only: true
+    immutable: true
+    description: |
+      The task Id of the task.
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    output: true
+    description: |
+      The relative resource name of the task, of the form: projects/{project_number}/locations/{locationId}/lakes/{lakeId}/ tasks/{name}.
+  - !ruby/object:Api::Type::String
+    name: 'uid'
+    output: true
+    description: |
+      System generated globally unique ID for the task. This ID will be different if the task is deleted and re-created with the same name.
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    output: true
+    description: |
+      The time when the task was created.
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    output: true
+    description: |
+      The time when the task was last updated.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      User-provided description of the task.
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      User friendly display name.
+  - !ruby/object:Api::Type::Enum
+    name: 'state'
+    output: true
+    description: |
+      Current state of the task.
+    values:
+      - :STATE_UNSPECIFIED
+      - :ACTIVE
+      - :CREATING
+      - :DELETING
+      - :ACTION_REQUIRED
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: 'labels'
+    description: |
+      User-defined labels for the task.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'triggerSpec'
+    required: true
+    description: |
+      Configuration for the cluster
+    properties:
+      - !ruby/object:Api::Type::Enum
+        name: 'type'
+        required: true
+        immutable: true
+        description: |
+          Trigger type of the user-specified Task
+        values:
+          - :TYPE_UNSPECIFIED
```

Comment:
```suggestion
```
For user supplied ENUM values we don't need to include unspecified status values

Reply:
> 

It is a required field and couldn't find any relevant documentation. It has been done similar to other product magic-modules.

Please provide a bit more context .

Reply:
It is a required and immutable user-entered field that is not computed by the API at any point -- so while the "UNSPECIFIED" value is configured into the enum on the API side, if it has no usage case over the other two options (which I'm assuming to be the case because it is not referenced in any user guides for Dataplex), it similarly does not need to be in the Terraform configuration because it will not be used by end-users.

Reply:
Resolved